test/Jsr292/src/com/ibm/j9/jsr292/api/MethodHandleAPI_dropArgumentsToMatch.java

```diff
@@ -0,0 +1,399 @@
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+package com.ibm.j9.jsr292.api;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+public class MethodHandleAPI_dropArgumentsToMatch {
+	public static class Hash_Methods {
+		public Hash_Methods() {}
+
+		public String hash_boolean(boolean a, boolean b) {
+			return String.valueOf(a && b);
+		}
+
+		public String hash_byte(byte a, byte b) {
+			return new String(new byte[] {a, b});
+		}
+
+		public String hash_char(char a, char b) {
+			return "" + a + b;
+		}
+
+		public String hash_int(int a, int b) {
+			return String.valueOf(a) + String.valueOf(b);
+		}
+
+		public String hash_double(double a, double b) {
+			return String.valueOf(a) + String.valueOf(b);
+		}
+
+		public String hash_array(int[] a, int[] b) {
+			String res = "";
+			for (int i = 0; i < a.length; i++) {
+				res += String.valueOf(a[i]);
+			}
+			for (int j = 0; j < b.length; j++) {
+				res += String.valueOf(b[j]);
+			}
+			return res;
+		}
+
+		public String hash_multi_arguments(String a, String b, String c, String d) {
+			return a + b + c + d;
+		}
+	}
+
+
+	/**
+	 * test for null argument
+	 * @throws NullPointerException
+	 */
+	@Test(groups = { "level.extended" }, expectedExceptions = NullPointerException.class)
+	public static void test_dropArgumentsToMatch_null_handle() throws Throwable {
+		MethodHandle h0 = MethodHandles.constant(boolean.class, true);
+
+		MethodHandle h1 = MethodHandles.dropArgumentsToMatch(null, 0, h0.type().parameterList(), 0);
+
+		Assert.fail("dropArgumentsToMatch method is nut supposed to accept null MethodHandle");
+	}
+
+	/**
+	 * test for null argument
+	 * @throws NullPointerException
+	 */
+	@Test(groups = { "level.extended" }, expectedExceptions = NullPointerException.class)
+	public static void test_dropArgumentsToMatch_null_list() throws Throwable {
+		MethodHandle h0 = MethodHandles.constant(boolean.class, true);
+
+		MethodHandle h1 = MethodHandles.dropArgumentsToMatch(h0, 0, null, 0);
+
+		Assert.fail("dropArgumentsToMatch method is nut supposed to accept null parameter list");
+	}
+
```

Comment:
A test with a target methodHandle that accepts no parameters (ie ()Ljava/lang/String;) might be usefull as well. 

Similarly a non-NULL empty parameter list would also be usefull